addon/index.js

```diff
@@ -44,7 +44,7 @@ export default Mixin.create({
     }
     const scrollElement = get(this, 'service.scrollElement');
 
-    const preserveScrollPosition = get(lastTransition, 'handler.controller.preserveScrollPosition');
+    const preserveScrollPosition = transitions.some((transition) => get(transition, 'handler.controller.preserveScrollPosition'));
```

Comment:
Is it possible to walk backwards on the transitions.  Would it make a difference here?

Reply:
@hbrysiewicz So if there are 5 transitions do we want to start at the last one and walk backwards until we find one with `handler.controller.preserveScrollPosition`?  This way will start at the earliest one.  What are your thoughts?

Reply:
Yeah you are right.  It's just a boolean value.  Ignore my comment!